src/Microsoft.SqlTools.ServiceLayer/EditData/EditDataService.cs

```diff
@@ -0,0 +1,328 @@
+
+                // Setup a callback for when the query failed to be created
+                Func<string, Task> queryCreateFailureCallback = requestContext.SendError;
+
+                // Setup a callback for when the query completes execution successfully
+                Query.QueryAsyncEventHandler queryCompleteSuccessCallback = async query =>
+                {
+                    EditSessionReadyParams readyParams = new EditSessionReadyParams
+                    {
+                        OwnerUri = initParams.OwnerUri
+                    };
+
+                    try
+                    {
+                        // Validate the query for a session
+                        ResultSet resultSet = Session.ValidateQueryForSession(query);
+
+                        // Get a connection we'll use for SMO metadata lookup (and committing, later on)
+                        DbConnection conn = await connectionService.GetOrOpenConnection(initParams.OwnerUri, ConnectionType.Edit);
+                        var metadata = metadataFactory.GetObjectMetadata(conn, resultSet.Columns, 
+                            initParams.ObjectName, initParams.ObjectType);
+
+                        // Create the session and add it to the sessions list
+                        Session session = new Session(resultSet, metadata);
+                        if (!ActiveSessions.TryAdd(initParams.OwnerUri, session))
+                        {
+                            throw new InvalidOperationException("Failed to create edit session, session already exists.");
+                        }
+                        readyParams.Success = true;
+                    }
+                    catch (Exception)
+                    {
+                        // Request that the query be disposed
+                        await queryExecutionService.InterServiceDisposeQuery(initParams.OwnerUri, null, null);
+                        readyParams.Success = false;
+                    }
+
+                    // Send the edit session ready notification
+                    await requestContext.SendEvent(EditSessionReadyEvent.Type, readyParams);
+                };
+
+                // Setup a callback for when the query completes execution with failure
+                Query.QueryAsyncEventHandler queryCompleteFailureCallback = query =>
+                {
+                    EditSessionReadyParams readyParams = new EditSessionReadyParams
+                    {
+                        OwnerUri = initParams.OwnerUri,
+                        Success = false
+                    };
+                    return requestContext.SendEvent(EditSessionReadyEvent.Type, readyParams);
+                };
+
+                // Put together a query for the results and execute it
+                ExecuteStringParams executeParams = new ExecuteStringParams
+                {
+                    Query = $"SELECT * FROM {SqlScriptFormatter.FormatMultipartIdentifier(initParams.ObjectName)}",
```

Comment:
>$ [](start = 28, length = 1)

Is this a new string.format syntax? We should have a plan for these new language features. They're nice, I just worry about how long it takes for people to learn these. We're often a bit conservative about how long before we adopt this kind of syntactic sugar in order to avoid .Net updates (not an issue here since it's done at compile time) but also ramp up time for people not familiar with latest & greatest.